aggregation/simple/src/main/java/com/spotify/heroic/aggregation/simple/TdigestMergingBucket.kt

```diff
@@ -36,20 +37,36 @@ import java.util.concurrent.atomic.AtomicReference
  *
  */
 data class TdigestMergingBucket(override val timestamp: Long) : AbstractBucket(), TDigestBucket {
-    private val datasketch : AtomicReference<TDigest> = TdigestInstanceUtils.buildAtomicReference()
-
+    private val datasketch : TDigest = TdigestInstanceUtils.inital()
 
     override fun updateDistributionPoint(key: Map<String, String>, sample : DistributionPoint) {
         val heroicDistribution : HeroicDistribution = HeroicDistribution.create(sample.value().value)
         val serializedDatasketch = heroicDistribution.toByteBuffer()
-        datasketch.getAndUpdate(TdigestInstanceUtils.getOp(serializedDatasketch))
+        val input: TDigest = MergingDigest.fromBytes(serializedDatasketch)
+        if ( input.size() > 0) {
+            update(input)
+        }
     }
 
+    @Synchronized fun update(input : TDigest) {
+        //This is a temp fix to handle corrupted datapoint.
```

Comment:
when does this happen?

Reply:
It happens when you query tracingapp data, but this is the only data we currently have. So it is not clear if it is a problem with the consumer pipeline or tracingapp.

 Note,  it is common for current histogram to send data to the consumer even if there is no recording in the datagram in the case of distribution this may be causing issue. We are not sure @samfadrigalan  and I are looking into it.

Reply:
interesting, thanks for the explanation 👍